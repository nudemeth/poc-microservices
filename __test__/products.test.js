import React from 'react';
import Router from 'next/router'
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import ConnectedProducts, { Products } from '../pages/products';

const mockedRouter = { prefetch: () => {} };
Router.router = mockedRouter;

describe('Products page', () => {
    it('Should show "Products Page: Prodcuts Page" in Products page', () => {
        const spyDispatch = sinon.spy();
        const products = shallow(<Products dispatch={spyDispatch} text='Products Page' />);
        expect(products.find('h1').text()).toEqual('Products Page: Products Page');
        sinon.assert.calledWith(spyDispatch, { type: "LOAD_PRODUCTS" });
    });
});

jest.mock('../components/page/products/ProductList', () => {
    return () => "<div>Mocked Product List</div>";
});

describe('Products page with Snapshot Testing', () => {
    it('Should show "Products Page: Prodcuts Page" in Products page', () => {
        const spyDispatch = sinon.spy();
        const products = shallow(<Products dispatch={spyDispatch} text='Products Page' />);
        const tree = toJSON(products);
        expect(tree).toMatchSnapshot();
    });
});