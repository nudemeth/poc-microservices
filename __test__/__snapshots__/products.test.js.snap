// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products page with Snapshot Testing Should show "Products Page: Prodcuts Page" in Products page 1`] = `
<Root
  title="Products"
>
  <h1>
    Products Page: 
    Products Page
  </h1>
  <Component />
</Root>
`;
